src/main/java/org/skriptlang/skript/bukkit/spark/expressions/ExprMSPT.java

```diff
@@ -0,0 +1,135 @@
+
+@Name("MSPT Usage")
+@RequiredPlugins("Paper 1.21+ or Spark")
+@Description({
+	"Returns the MSPT (milliseconds per tick) readings from the last 10 seconds, 1 minute and 5 minutes.",
+	"This expression can only be used if the server has Spark or you have Spark installed as a plugin."
+})
+@Examples({
+	"broadcast all server tick times"
+})
+public class ExprMSPT extends SimpleExpression<Number> {
+
+	private int index;
+	private String expr = "server tick";
+	private static Spark spark;
+	private static int msptType;
+
+	static {
+		if (Skript.classExists("me.lucko.spark.api.SparkProvider")) {
+			Skript.registerExpression(ExprMSPT.class, Number.class, ExpressionType.SIMPLE,
+				"[the] (server tick (duration|time)|mspt) (from|of) [the] last (10|ten) seconds using (:min[imum]|:mean|:median|:95th percentile|:max[imum]|:all)",
+				"[the] (server tick (duration|time)|mspt) (from|of) [the] last [(1|one)] minute using (:min[imum]|:mean|:median|:95th percentile|:max[imum]|:all)",
+				"[the] (server tick (duration|time)|mspt) (from|of) [the] last (5|five) minutes using (:min[imum]|:mean|:median|:95th percentile|:max[imum]|:all)",
+				"all [the] (server tick (durations|times)|mspt values)");
+		}
+	}
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		spark = SparkProvider.get();
+		expr = parseResult.expr;
+		index = matchedPattern;
+		msptType = parseResult.mark;
+		return true;
+	}
+
+	@Override
+	protected Number @Nullable [] get(Event event) {
+		spark = SparkProvider.get();
+		return getMSPTStats(index);
+	}
+
+	@Override
+	public boolean isSingle() {
+		return index != 3;
+	}
+
+	@Override
+	public Class<? extends Number> getReturnType() {
+		return Number.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return expr + " from the last " + index;
+	}
+
+	private static Number[] getMSPTStats(int index) {
+		return switch (index) {
+			case 0 -> getMSPTForWindow(StatisticWindow.MillisPerTick.SECONDS_10);
+			case 1 -> getMSPTForWindow(StatisticWindow.MillisPerTick.MINUTES_1);
+			case 2 -> getMSPTForWindow(StatisticWindow.MillisPerTick.MINUTES_5);
+			default -> getAllMSPTStats();
+		};
+	}
+
+	private static Number[] getMSPTForWindow(StatisticWindow.MillisPerTick window) {
+		GenericStatistic<DoubleAverageInfo, StatisticWindow.MillisPerTick> mspt = spark.mspt();
+		DoubleAverageInfo info = mspt.poll(window);
+		return switch (msptType) {
+			case 1 -> new Number[]{info.min()};
+			case 2 -> new Number[]{info.mean()};
+			case 3 -> new Number[]{info.median()};
+			case 4 -> new Number[]{info.percentile95th()};
+			case 5 -> new Number[]{info.max()};
+			default -> getAllMSPTStatsPerWindow(window);
+		};
+	}
+
+	private static Number[] getAllMSPTStatsPerWindow(StatisticWindow.MillisPerTick window) {
+		GenericStatistic<DoubleAverageInfo, StatisticWindow.MillisPerTick> mspt = spark.mspt();
```

Comment:
you shouldn't be getting this every time
you can just get it once as a static var after you get the spark instance.